### PR TITLE
Fix source_tree and relative path

### PR DESCRIPTION
**Pull request by wuhao5**
https://github.com/google/EarlGrey/issues/510

### CODE REVIEW

**Comment by tirodkar** [comment]
@bootstraponline, do test this out if you can for Carthage!

File: gem/lib/earlgrey/version.rb [hunk 1]
```diff
@@ -14,5 +14,5 @@
 #  limitations under the License.
 
 module EarlGrey
-  VERSION = '0.2.1'.freeze unless defined? ::EarlGrey::VERSION
+  VERSION = '0.2.2'.freeze unless defined? ::EarlGrey::VERSION
```
Comment (bootstraponline) [inline gem/lib/earlgrey/version.rb]: 😭

**Comment by bootstraponline** [comment]
> @bootstraponline, do test this out if you can for Carthage!

I'll test it out now.

File: gem/lib/earlgrey/version.rb (hunk 1 above)
Comment (wuhao5) [inline gem/lib/earlgrey/version.rb]: Hah, sorry I forgot what was the decision to be made, is it to be sync'd or not?

Note that we are doing more mainstream releases than gem releases.

Comment (bootstraponline) [inline gem/lib/earlgrey/version.rb]: can we just make it `1.9.2`? `0.2.2` is not conveying any useful info to the end user

**Comment by bootstraponline** [comment]
I verified this works on the latest version of carthage.

```
gem build earlgrey.gemspec
gem list earlgrey
gem install --local earlgrey-0.2.2.gem
gem list earlgrey
cd Demo/EarlGreyExample
brew update
brew upgrade carthage
earlgrey install --carthage --swift-version=3.0 --target=EarlGreyExampleSwiftTests
carthage update EarlGrey --platform ios
open EarlGreyExample.xcodeproj
```

**Comment by khandpur** [comment]
lots of diff detected here: https://travis-ci.org/google/EarlGrey/jobs/235431968

**Comment by khandpur** [comment]
@bootstraponline thanks for testing!

**Comment by khandpur** [comment]
@wuhao could we perhaps add a test for this with R.swift? That way we can catch regressions in the future?

**Comment by wuhao5** [comment]
@khandpur looks like you mentioned the wrong person :)

forgot to run rake test, good catch, will also add those tests too. the only concern is if we should support R.swift and add another dependency here. I'd say R.swift is doing something incompatible with the unusual paths which have some special characters. R.swift should ignore or warn, rather than just fail.

File: gem/lib/earlgrey/version.rb (hunk 1 above)
Comment (wuhao5) [inline gem/lib/earlgrey/version.rb]: I am dropping this now and update it with the release schedule.

**Comment by wuhao5** [comment]
Fixed the fixtures and added R.swift to the example project. waiting for CI to finish.

File: Demo/EarlGreyExample/Podfile [hunk 2]
```diff
@@ -30,6 +30,9 @@ end
 target "EarlGreyExampleSwiftTests" do
   project 'EarlGreyExample'
 
+  # Make sure it is compatible with other pods, i.e. R.swift
+  pod 'R.swift'
```
Comment (tirodkar) [inline Demo/EarlGreyExample/Podfile]: Do we have any R.swift code in this target? We use EGExample / EGExampleSwift for Sanity Testing our new pods. Might not want to add a new dependency on it.

Comment (tirodkar) [inline Demo/EarlGreyExample/Podfile]: I'd create a new target or add it to the Carthage tests or so? Maybe run some R.swift code to it?

Comment (wuhao5) [inline Demo/EarlGreyExample/Podfile]: reverted this for now.

**Comment by tirodkar** [comment]
Waiting on CI. Thanks.

**Pull request merged**